go/test/endtoend/tabletmanager/throttler/throttler_test.go

```diff
@@ -156,19 +175,44 @@ func TestLag(t *testing.T) {
 		time.Sleep(2 * time.Second)
```

Comment:
Are these durations used in the tests (2 seconds, 10 seconds, 5 seconds) based on values configured elsewhere? Just wondering if these can become flaky if a configuration is changed elsewhere.

Reply:
Excellent question. There are some hard coded intervals here:

https://github.com/vitessio/vitess/blob/627f0923af87302250476970f8e80cf91d83e5d7/go/vt/vttablet/tabletserver/throttle/throttler.go#L37-L41

Basically:
- after a cluster starts, it takes ~5sec for throttler to know about all replicas _and_ collect data from those replicas _and_ aggregate that data -- meaning an API `/throttler/check` can provide a reliable answer. Hence waiting for `10sec` (can reduce that to e.g.`7s` but I feel `10sec` is much safer against flakyness.
- user apps will cache throttler results for some `250ms` (e.g. see in the vreplication PRs), so a `1sec` sleep can ensure the cache is cleared
- The default lag threshold is `1s`, hence the `2 * time.Second` sleep after `StopReplication()`, to make sure when we next check for lag, there _is_ a throttle-grade lag. Having said that, you are right that this is overridable -- so I just added
```
			"-throttle_threshold", "1s",
```
  to this test's `VtTabletExtraArgs` to ensure the threshold is `1s`.
- The sleep for `5 * time.Second` after `StartReplication` is just heuristic to allow the replication to catch up, and does not depend on the throttler configuration. Again, catch up is likely to happen in less than `1s` but I feel `5s` is great against flakyness.
- - The sleep for `10 * time.Second` after `ChangeTabletType` is because it will take that amount of time for the throttler to identify the new roster. It's hard coded in `mysqlRefreshInterval        = 10 * time.Second`, and now I've actually uppsed the test to sleep for `12 * time.Second` to avoid flakyness.

I've moreover now made these numbers as constants in the test. Now the waits are named, and its clearer what each wait means.

Reply:
new code:
```go
const (
	throttlerInitWait        = 10 * time.Second
	accumulateLagWait        = 2 * time.Second
	mysqlRefreshIntervalWait = 12 * time.Second
	replicationCatchUpWait   = 5 * time.Second
)
```
...
```go
		time.Sleep(mysqlRefreshIntervalWait)
```